File: spikes/Spike3/MenuItem.cpp
/*#include "MenuItem.h"

void MenuItem::addOption(Option* newOption)
{
   options.push_back(newOption);
}

void MenuItem::removeOption(int index)
{
   options.erase(options.begin() + index);
}

void MenuItem::updateName(string newName)
{
   name = newName;
}

void MenuItem::updateDescription(string newDesc)
{
   description = newDesc;
}

void MenuItem::updateCalories(int newCal)
{
   calories = newCal;
}

void MenuItem::updatePrice(float newPrice)
{
   price = newPrice;
}

void MenuItem::updatetimeCost(float newTime)
{
   timeCost = newTime;
}

Option* MenuItem::getOption(string name)
{
   for (int i = 0; i < options.size(); i++)
      if (options[i]->getName() == name) return options[i];
   return NULL;
}

int MenuItem::getCalories()
{
   return calories;
}

string MenuItem::getName()
{
   return name;
}

float MenuItem::getPrice()
{
   return price;
}

float MenuItem::getTimeCost()
{
   return timeCost;
}

string MenuItem::getDescription()
{
   return description;
}

int MenuItem::getOptionsListSize()
{
   return options.size();
}

bool MenuItem::operator< (const MenuItem & compareFrom) const
{
   return name < compareFrom.name;
}*/

int main()
{
   return 0;
}

File: spikes/spike4/RestaurantKiosk/ObjectClasses/MenuItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RestaurantKiosk
{
   public class MenuItem
   {
      private bool availible;
      private int calories;
      private string description;
      private string name;
      private List<Option> options;
      private float price;
      private float timeCost;

      public MenuItem()
      {
         availible = false;
         calories = 0;
         description = "";
         name = "";
         price = 0.0f;
         timeCost = 0.0f;
      }

      public MenuItem(int cal, string desc, string itemName, float itemPrice, float itemTimeCost)
      {
         calories = cal;
         description = desc;
         name = itemName;
         price = itemPrice;
         timeCost = itemTimeCost;
         availible = true;
      }

      //MenuItem(const MenuItem & cpy) : calories(cpy.calories), description(cpy.description), name(cpy.name), price(cpy.price), timeCost(cpy.timeCost), availible(cpy.availible) { }

      public void addOption(Option newOption)
      {
         options.Add(newOption);
      }

      public void removeOption(int index)
      {
         options.RemoveAt(index);
      }

      public void updateName(string newName)
      {
         name = newName;
      }

      public void updateDescription(string newDesc)
      {
         description = newDesc;
      }

      public void updateCalories(int newCal)
      {
         calories = newCal;
      }

      public void updatePrice(float newPrice)
      {
         price = newPrice;
      }

      public void updatetimeCost(float newTime)
      {
         timeCost = newTime;
      }

      public Option getOption(string name)
      {
         foreach (var option in options)
         {
            if (option.getName().Equals(name, StringComparison.InvariantCultureIgnoreCase))
            {
               return option;
            }
         }
         return null;
      }

      public int getCalories()
      {
         return calories;
      }

      public string getName()
      {
         return name;
      }

      public float getPrice()
      {
         return price;
      }

      public float getTimeCost()
      {
         return timeCost;
      }

      public string getDescription()
      {
         return description;
      }

      public int getOptionsListSize()
      {
         return options.Count;
      }

   }
}
